### PR TITLE
FeatureBadge: Update FeatureBadge to support current release stages

### PR DESCRIPTION
**Pull request by ivanahuckova**
Aligning with our documentation on the release lifecycle (https://grafana.com/docs/release-life-cycle/), this PR updates the `FeatureBadge` component to support the release stages we actively maintain, which are `experimental` and `preview` (shared for private and public preview). When a feature reaches `general availability` (GA), it no longer requires a badge, so I haven't added that one). 

Additionally, I have deprecated the `alpha` and `beta` states. This change ensures that the `FeatureBadge` reflects only the relevant stages of our feature release process.

image

### CODE REVIEW

File: packages/grafana-ui/src/components/InfoBox/FeatureInfoBox.tsx
Comment (tskarhed) [inline packages/grafana-ui/src/components/InfoBox/FeatureInfoBox.tsx:69]: The assumption here is that we don't need a default case since the featureState isn't optional?

Comment (ivanahuckova) [reply packages/grafana-ui/src/components/InfoBox/FeatureInfoBox.tsx:69]: Yeah. Also I am not sure why the code was structured that way - because as you've said, the `featureState` is required and it could have been only `beta` or `alpha`. So in my case, I am just adding a `switch case` for every available option.

**Review by tskarhed — CHANGES_REQUESTED** [review]
`InfoBox` is a deprecated component, but `FeatureBadge` seems to be in use, so maybe it would be good to put in a separate folder, file and story?

**Comment by ivanahuckova** [comment]
Moved to separate folder with separate story in https://github.com/grafana/grafana/pull/87555/commits/c96add6583b2834a22531efbc5aa320d93a2bb1c

File: packages/grafana-data/src/types/app.ts
Comment (jackw) [inline packages/grafana-data/src/types/app.ts:130]: I could be wrong here but isn't this related to the [plugins.alpha](https://github.com/grafana/grafana/blob/main/conf/defaults.ini#L1592) config setting? 🤔

Comment (tskarhed) [reply packages/grafana-data/src/types/app.ts:130]: That seems to make sense. In that case, maybe it should show up as `Experimental` in the UI instead? Although it will lead to confusion in the config, but be more consistent with the Grafana release life cycle.

Possibly that a mapping between `Alpha -> Experimental` and `Beta -> Preview`.

File: packages/grafana-data/src/types/app.ts
Comment (tskarhed) [inline packages/grafana-data/src/types/app.ts:134]: What's the reasoning for sharing public and private preview?

Comment (ivanahuckova) [reply packages/grafana-data/src/types/app.ts:134]: Not sure if I understand the question? We would for example use the Preview badge for features that are currently in `private` or `public` preview, to mark them.

File: packages/grafana-data/src/types/app.ts
Comment (ivanahuckova) [reply packages/grafana-data/src/types/app.ts:130]: When I looked how this is used, I haven't seen anything related to plugin/dependency on this setting:  https://github.com/search?q=org%3Agrafana+%22FeatureState.%22&type=code

File: packages/grafana-data/src/types/app.ts
Comment (ivanahuckova) [reply packages/grafana-data/src/types/app.ts:134]: Based on what was communicated to me by @stevendungan and @zizzpudding we don't use `alpha` `beta` to mark features for a while now. The language has changed to `experimental` and `preview`.  

@mjseaman could you please provide guidance on this.

Comment (tskarhed) [reply packages/grafana-data/src/types/app.ts:134]: Ah, sorry for not being clear. My reasoning is why don't we have two separate `Private preview` and `Public preview`?

File: packages/grafana-data/src/types/app.ts
Comment (tskarhed) [reply packages/grafana-data/src/types/app.ts:130]: That's true, it might just use a `Badge`

File: packages/grafana-data/src/types/app.ts
Comment (ivanahuckova) [reply packages/grafana-data/src/types/app.ts:134]: Oh! So we (obs logs) have been using 'Preview' for both, as it is not relevant to the user whether the preview is public or private. Also, 'Private/Public Preview' is quite lengthy.

Comment (tskarhed) [reply packages/grafana-data/src/types/app.ts:134]: I guess so. If you look at the feature docs you can find the "real" type of preview. I am fine with okaying this, but if you want to you can verify with the product people.

**Review by tskarhed — APPROVED** [review]
This is okay from my side (thanks for moving around files and stuff!). If you want to, you can verify with product people if they have any opinions around using `Public/private preview` as separate options. I think it doesn't matter to the end user, and the people who care enough can find the "real" state of it.

**Comment by ivanahuckova** [comment]
@tskarhed thank you! I'll wait on approval from PMs to be 💯 aligned with FeatureBadge wording.

**Pull request merged**
